### PR TITLE
[Hub Generated] Review request for Microsoft.ContainerService/aks to add version preview/2023-11-02-preview

### PR DESCRIPTION
**Pull request by FumingZhang**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://openapihub.azure-devex-tools.com/branch/Azure/azure-rest-api-specs/FumingZhang-containerservice-Microsoft.ContainerService-2023-11-02-preview...main/).
# ARM (Control Plane) API Specification Update Pull Request 

> Tip: overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

diagram

<details>
<summary> Click here to see the details of Step 1 </summary>

### Breaking changes review (Diagram Step 1)

If the automation determines you have breaking changes, i.e. Step 1 from the diagram applies to you,
  you must follow the [breaking changes process].  
  **IMPORTANT** This applies even if:
- The tool fails while it shouldn't, e.g. due to runtime exception, or incorrect detection of breaking changes.
- You believe there is no need for you to request breaking change approval, for any reason. 
    Such claims must be reviewed, and the process is the same.

</details>

<details> 
<summary> Click here to see the details of Step 2 </summary>

### ARM API changes review (Diagram Step 2)

- If this PR is in purview of ARM review then automation will add the `ARMReview` label.
- If you want to force ARM review, add the label yourself.
- Proceed according to the diagram at the top of this comment.

</details>

<details>
<summary> Click here to see the diagram footnotes</summary>

### Diagram footnotes

[1] [ARM review queue] (for **merge** queues, see [2])  
The PRs are processed by time opened, ascending. Your PR may show up on 2nd or later page. 
If you addressed Step 1 from the diagram and your PR is not showing up in the queue, ensure the label `ARMChangesRequested` 
is removed from your PR. This should cause the label `WaitForARMFeedback` to be added.
[2] [public repo merge queue], [private repo merge queue] (for **ARM review** queue, [1])

If you need further help with anything, see `Getting help` section below.

</details>

## Purpose of this PR

What's the purpose of this PR? Check all that apply. This is **mandatory**!

  - [x] New API version. (If API spec is not defined in TypeSpec, the PR should have been generated using [OpenAPI Hub](https://aka.ms/openapihub)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix swagger quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the Diagram Step 2, "ARM API changes Review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://aka.ms/azsdk/pr-suppressions) 
to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

[ARM review queue]: https://github.com/search?q=org%3AAzure+is%3Apr+is%3Aopen+label%3AWaitForARMFeedback+-label%3AIDCDevDiv++draft%3Afalse+sort%3Acreated-asc+&type=pullrequests
[public repo merge queue]: https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+draft%3Afalse+sort%3Acreated-asc
[private repo merge queue]: https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AApproved-OkToMerge+draft%3Afalse+sort%3Acreated-asc
[breaking changes process]: https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/launchingproductbreakingchanges#breaking-change-process-1

### CODE REVIEW

**Comment by FumingZhang** [comment]
@rkmanda, I would like to apply for suppression of failed CI [Swagger Avocado](https://github.com/Azure/azure-rest-api-specs/pull/27431/checks?check_run_id=20643853615) as we deliberately did not add the new swagger file [aks/preview/2023-11-02-preview/scheduledEvents.json](https://github.com/Azure/azure-rest-api-specs/blob/5aefecbe32408d68e731023417158b347cc0c1f2/specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-11-02-preview/scheduledEvents.json) (change committed in #26774) to the SDK config. The intended consumer of this swagger model is the Azure Resource Notification team. Their requirement is the swagger should be published publicly as the consumption model, but meanwhile AKS does not want to expose it to the SDK.

\+ avocado owners @konrad-jamrozik, @raych1 
\+ feature owners @JunSun17, @matthchr 

image
Followed the guidance in doc https://eng.ms/docs/products/azure-developer-experience/design/specs-pr-guides/pr-suppressions

**Comment by konrad-jamrozik** [comment]
@FumingZhang don't worry about avocado, it is non-blocking. Instead, see `Getting help` section from your PR description.

**Comment by raosuhas** [comment]
>         }

consider updating these to enum as explained here : 

[https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md](https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md)

---
Refers to: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-11-02-preview/managedClusters.json:7243 in 5aefecb. [](commit_id = 5aefecbe32408d68e731023417158b347cc0c1f2, deletion_comment = False)

**Comment by raosuhas** [comment]
>           "type": "boolean",

same here: 

for example :

Version: Default (later you could add Latest\Oldest etc as required)

---
Refers to: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-11-02-preview/managedClusters.json:7648 in 5aefecb. [](commit_id = 5aefecbe32408d68e731023417158b347cc0c1f2, deletion_comment = False)

File: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-11-02-preview/managedClusters.json [hunk 1]
```diff
@@ -0,0 +1,9438 @@
...
+      "x-ms-enum": {
+        "name": "OSSKU",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "Ubuntu",
+            "description": "Use Ubuntu as the OS for node images."
+          },
+          {
+
```
Comment (FumingZhang) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-11-02-preview/managedClusters.json:7244]: > > ```
> >     }
> > ```
> 
> consider updating these to enum as explained here :
> 
> https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md
> 
> Refers to: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-11-02-preview/managedClusters.json:7243 in [5aefecb](https://github.com/Azure/azure-rest-api-specs/commit/5aefecbe32408d68e731023417158b347cc0c1f2). [](commit_id = [5aefecb](https://github.com/Az> > ```
> >     }
> > ```
> 
> consider updating these to enum as explained here :
> 
> https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md
> 
> Refers to: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-11-02-preview/managedClusters.json:7243 in [5aefecb](https://github.com/Azure/azure-rest-api-specs/commit/5aefecbe32408d68e731023417158b347cc0c1f2). [](commit_id = [5aefecb](https://github.com/Azure/azure-rest-api-specs/commit/5aefecbe32408d68e731023417158b347cc0c1f2), deletion_comment = False)

ure/azure-rest-api-specs/commit/5aefecbe32408d68e731023417158b347cc0c1f2), deletion_comment = False)

Comment (FumingZhang) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-11-02-preview/managedClusters.json:7243]: > > ```
> >     }
> > ```
> 
> consider updating these to enum as explained here :
> 
> https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md
> 
> Refers to: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-11-02-preview/managedClusters.json:7243 in [5aefecb](https://github.com/Azure/azure-rest-api-specs/commit/5aefecbe32408d68e731023417158b347cc0c1f2). [](commit_id = [5aefecb](https://github.com/Azure/azure-rest-api-specs/commit/5aefecbe32408d68e731023417158b347cc0c1f2), deletion_comment = False)

**Comment by FumingZhang** [comment]
> > ```
> >     }
> > ```
> 
> consider updating these to enum as explained here :
> 
> https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md
> 
> Refers to: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-11-02-preview/managedClusters.json:7243 in [5aefecb](https://github.com/Azure/azure-rest-api-specs/commit/5aefecbe32408d68e731023417158b347cc0c1f2). [](commit_id = [5aefecb](https://github.com/Azure/azure-rest-api-specs/commit/5aefecbe32408d68e731023417158b347cc0c1f2), deletion_comment = False)

changed added in PR #26850

**Comment by FumingZhang** [comment]
> > ```
> >       "type": "boolean",
> > ```
> 
> same here:
> 
> for example :
> 
> Version: Default (later you could add Latest\Oldest etc as required)
> 
> Refers to: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-11-02-preview/managedClusters.json:7648 in [5aefecb](https://github.com/Azure/azure-rest-api-specs/commit/5aefecbe32408d68e731023417158b347cc0c1f2). [](commit_id = [5aefecb](https://github.com/Azure/azure-rest-api-specs/commit/5aefecbe32408d68e731023417158b347cc0c1f2), deletion_comment = False)

change added in PR #26798

**Comment by NickKeller** [comment]
> > ```
> >       "type": "boolean",
> > ```
> 
> same here:
> 
> for example :
> 
> Version: Default (later you could add Latest\Oldest etc as required)
> 
> Refers to: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-11-02-preview/managedClusters.json:7648 in [5aefecb](https://github.com/Azure/azure-rest-api-specs/commit/5aefecbe32408d68e731023417158b347cc0c1f2). [](commit_id = [5aefecb](https://github.com/Azure/azure-rest-api-specs/commit/5aefecbe32408d68e731023417158b347cc0c1f2), deletion_comment = False)

@raosuhas this property is absolutely a boolean. Either it is the default version, or it isn't. We have no plans to add a "Latest" or "Oldest" category

**Comment by matthchr** [comment]
> > ```
> >     }
> > ```
> 
> consider updating these to enum as explained here :
> 
> https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md
> 
> Refers to: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-11-02-preview/managedClusters.json:7243 in [5aefecb](https://github.com/Azure/azure-rest-api-specs/commit/5aefecbe32408d68e731023417158b347cc0c1f2). [](commit_id = [5aefecb](https://github.com/Azure/azure-rest-api-specs/commit/5aefecbe32408d68e731023417158b347cc0c1f2), deletion_comment = False)

These two new boolean fields are made to match the fields exposed by VMSS. Those fields are shown [here](https://learn.microsoft.com/en-us/rest/api/compute/virtual-machine-scale-sets/create-or-update?view=rest-compute-2023-10-02&tabs=HTTP#uefisettings).

Since these fields are just proxying VMSS fields, which themselves are booleans, it doesn't make sense to use an enum.

**Comment by FumingZhang** [comment]
Hey @msyyc, could you please help review and approve the `CI-BreakingChange-Python-Track2`? The latest released SDK is taking a stable API version as default and all listed added features in the comparison results are new in the preview API version compared to the stable API version, which is also in line with expectations.

https://github.com/Azure/azure-rest-api-specs/pull/27431/checks?check_run_id=20886076235

**Comment by FumingZhang** [comment]
Hey @weidongxu-microsoft, could you please help find out why CI [SDK azure-sdk-for-java](https://github.com/Azure/azure-rest-api-specs/pull/27431/checks?check_run_id=20886076609) is failing? I reran the CI but met the same error. The error message is
> ENOENT: no such file or directory, stat 'azure-sdk-for-java/sdk/containerservicemicrosoftcontainerserviceaks/azure-resourcemanager-containerservicemicrosoftcontainerserviceaks-generated/pom.xml'

**Review by visingla-ms — APPROVED** [review]

**Pull request merged**